File: OC_麻豆/OC_麻豆/ViewController.m
//
//  ViewController.m
//  OC_麻豆
//
//  Created by JM Zhao on 2017/5/22.
//  Copyright © 2017年 JunMing. All rights reserved.
//

#import "ViewController.h"
#import "UIButton+ButtonExtension.h"
#import "IndextableView.h"

#define SCREEN_WIDTH [UIScreen mainScreen].bounds.size.width
#define SCREEN_HEIGHT [UIScreen mainScreen].bounds.size.height

#define functionHeaderViewHeight 95
#define singleAppHeaderViewHeight 60

#define topOffsetY 155

@interface ViewController ()<UIScrollViewDelegate, UIGestureRecognizerDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // uiscrollView
    CGFloat height = SCREEN_HEIGHT - 64;
    UIScrollView *mainScrollView = [[UIScrollView alloc] initWithFrame:CGRectMake(0, 64, SCREEN_WIDTH, height)];
    mainScrollView.delegate = self;
    mainScrollView.contentSize = CGSizeMake(SCREEN_WIDTH, 100);
    mainScrollView.scrollIndicatorInsets = UIEdgeInsetsMake(155, 0, 0, 0);
    [self.view addSubview:mainScrollView];
    
    // navView
    UIView *navView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, SCREEN_WIDTH, 64)];
    navView.backgroundColor = [UIColor colorWithRed:65/255.0 green:128/255.0 blue:1 alpha:1];
    [self.view addSubview:navView];
    
    // mainNavView
    UIView *mainNavView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, SCREEN_WIDTH, 64)];
    mainNavView.backgroundColor = [UIColor colorWithRed:65/255.0 green:128/255.0 blue:1 alpha:1];
    [self.view addSubview:mainNavView];
    
    UIButton *pa
    
    // uiscrollView
    
    
    // uiscrollView
    
    
    // uiscrollView
    
    
    // uiscrollView
    
    
    // uiscrollView
    
    
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Swift_Enumerate/Swift_Enumerate/ViewController.swift
//
//  ViewController.swift
//  Swift_Enumerate
//
//  Created by JM Zhao on 2017/6/8.
//  Copyright © 2017年 奕甲智能 Oneplus Smartware. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {

    var tableView:UITableView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let button = UIButton.init(type: UIButtonType.system)
        button.setTitle("第一个", for: UIControlState.normal)
        button.addTarget(self, action: #selector(pushNext(sender:)), for: UIControlEvents.touchUpInside)
        button.frame = CGRect(x: 0, y: 0, width: 64, height: 64)
        button.center = self.view.center
        self.view.addSubview(button)
        
    }

    func creatTableView() {
        
        self.tableView = UITableView(frame: self.view.bounds, style: UITableViewStyle.plain)
        self.tableView?.delegate = self
        self.tableView?.dataSource = self
        self.view.addSubview(self.tableView!)
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return 10
        
    }
    
    func numberOfSections(in tableView: UITableView) -> Int {
        
        return 5
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        
    }
    
    func pushNext(sender:UIButton) {
        
        let fourVC = FourViewController()
        self.navigationController?.pushViewController(fourVC, animated: true)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

